homeassistant/helpers/script.py

```diff
@@ -420,6 +421,41 @@ async def async_stop(self) -> None:
         self._stop.set()
         await self._stopped.wait()
 
+    def _handle_exception(
+        self, exception: Exception, continue_on_error: bool, log_exceptions: bool
+    ) -> None:
+        if not isinstance(exception, (_AbortScript, _StopScript)) and log_exceptions:
+            self._log_exception(exception)
+
+        if not continue_on_error:
+            raise exception
+
+        # An explicit request to stop the script has been raised.
+        if isinstance(exception, _StopScript):
```

Comment:
Isn't this case covered by line 453?
```if not isinstance(exception, exceptions.HomeAssistantError)```

Reply:
Yes, this is just an early return (as instance calls are relatively expensive)

Reply:
It's doing an opposite check, but on difference exception types. Unless I'm misunderstanding, a `_StopScript` can't also be a `HomeAssistantError`, no?

Would this reduce the number of `isinstance` checks in the default flow when `continue_on_error == False`?
```python
 def _handle_exception(
        self, exception: Exception, continue_on_error: bool, log_exceptions: bool
    ) -> None:
        if not isinstance(exception, (_AbortScript, _StopScript)) and log_exceptions:
            self._log_exception(exception)

        if continue_on_error:
            # Only Home Assistant errors can be ignored.
            if isinstance(exception, exceptions.HomeAssistantError):
                # These are incorrect scripts, and not runtime errors that need to
                # be handled and thus cannot be stopped by `continue_on_error`.
                if not isinstance(
                    exception,
                    (
                        vol.Invalid,
                        exceptions.TemplateError,
                        exceptions.ServiceNotFound,
                        exceptions.InvalidEntityFormatError,
                        exceptions.NoEntitySpecifiedError,
                        exceptions.ConditionError,
                    ),
                ):
                    # Ignore the error
                    script_execution_set("error")
                    return

        raise exception
```

Reply:
There is no need to nest so deep, we generally handle guards/early returns.